:Clap install-binary! don't work on Windows

If maple is already available (once downloaded) it will be started by Clap immediatly (at any startup routine, haven't examined). So maple.exe (for windows) is already running as process, therefore the downloaded (new version) can not been saved on the location (because the OS has a write lock on the running file),

Furthermore the plugin says the download was successful and the user can not examine the error log bacuse the split window has been closed so fast, you had no chance to read the message.

Solution:
If updating binary, temporary stop maple job,

The problem I start do describe about locks on file also exists when comoiling by source:

```
= note: LINK : fatal error LNK1104: cannot open file 'C:\Users\jstocker9330\vimfiles\bundle\vim-clap\target\release\deps\maple.exe'
```

It is the same reason, there is a process ob "maple rpc" which will start as soon as you call any :Clap action. This keeps open as process while vim is running. A running process "locks" that file (at least on windows) so you cannot overwrite it.

Therefore we need to stop that "maple rpc" job (you spawn somewhere early) BEFORE any update of maple and after that restart it,

--- a/autoload/clap/installer.vim
+++ b/autoload/clap/installer.vim
@@ -17,6 +17,9 @@ function! s:run_term(cmd, cwd, success_info) abort
       execute 'silent! bd! '.bufnr
       call clap#helper#echo_info(a:success_info)
     endif
+    if clap#maple#is_available()
+      call clap#job#daemon#start(function('clap#client#handle'))
+    endif
   endfunction
 
   if has('nvim')
@@ -106,8 +109,13 @@ endfunction
 
 function! s:do_download() abort
   if !exists('s:current_version')
-    let version_line = readfile(s:maple_cargo_toml)[:5][-1]
-    let s:current_version = str2nr(matchstr(version_line, '0.1.\zs\d\+'))
+    if executable(s:prebuilt_maple_binary)
+      "Since v0.7
+      let version_line = system(s:prebuilt_maple_binary.' version')
+      let s:current_version = str2nr(matchstr(version_line, '0.1.\zs\d\+'))
+    else
+      let s:current_version = -1
+    endif
   endif
   " Since v0.14 maple itself is able to download the latest release binary.
   if executable(s:prebuilt_maple_binary) && s:current_version >= 14
@@ -123,6 +131,10 @@ function! clap#installer#force_download() abort
 endfunction
 
 function! clap#installer#install(try_download) abort
+  if clap#job#daemon#is_running()
+    call clap#job#daemon#stop()
+  endif
+
   " Always prefer to compile it locally.
   if executable('cargo')
     call clap#installer#build_all()

--- a/autoload/clap/installer.vim
+++ b/autoload/clap/installer.vim
@@ -118,7 +118,7 @@ function! s:do_download() abort
     endif
   endif
   " Since v0.14 maple itself is able to download the latest release binary.
-  if executable(s:prebuilt_maple_binary) && s:current_version >= 14
+  if s:current_version >= 14
     let cmd = s:prebuilt_maple_binary.' upgrade --download'
     call s:run_term(cmd, s:plugin_root_dir, 'download the latest prebuilt maple binary successfully')
   else

